### DISCUSSION TITLE
Can I use the DevTools Protocol?

### DISCUSSION THREAD

**Discussion by shouhujishu** [cefsharp/CefSharp#4244] [General]
Want to inject javascript function code perfectly through use the DevTools Protocol Page.addScriptToEvaluateOnNewDocument,
Currently IRenderProcessMessageHandler method is invalid for <iframe>.

**Comment by amaitland** [comment]
In future please use discussions to ask questions. https://github.com/cefsharp/CefSharp/discussions

Yes it's possible to use the DevTools protocol see https://github.com/cefsharp/CefSharp/issues/3158 and https://github.com/cefsharp/CefSharp/issues/3165

**Reply by shouhujishu** [reply]
Hello, thank you for your precious time, I want to into all the iframe frame, but I find that Page. The addScriptToEvaluateOnNewDocument in iframe domain name only match with the top-level domains, across the domain name is invalid.

Are there any other ways to tune this?

**Reply by amaitland** [reply]
> but I find that Page. The addScriptToEvaluateOnNewDocument in iframe domain name only match with the top-level domains, across the domain name is invalid.

The behavior is implement in chromium and not something we can control directly. You could possibly try disabling chromium site isolation. Try the https://peter.sh/experiments/chromium-command-line-switches/#disable-site-isolation-trials command line argument. 

Alternatively is FrameLoadStart called for the iframe you are referring to?
